<launch>

  <!-- Launch Apriltag stuff -->
  <include file="$(find the_mighty_sawyer)/launch/continuous_detection.launch" />

  <node name="rviz" pkg="rviz" type="rviz" args="-d $(find the_mighty_sawyer)/rviz/web.rviz" />
  <node name="apriltag_pose" pkg="the_mighty_sawyer" type="apriltag_pose" output="screen"/>

  <!-- Launch Image Detection stuff -->
  <node name="camera_converter" pkg="the_mighty_sawyer" type="camera_converter"/>
    <node name="find_object_2d" pkg="find_object_2d" type="find_object_2d" args="image:=image_converted">
        <param name="~gui" value="false"/>
        <param name="~objects_path" value="$(find the_mighty_sawyer)/images/vision" />
    </node>
  <node name="grab_bag_server" pkg="the_mighty_sawyer" type="grab_bag_server"/>

  <!-- Launch Movement Stuff -->
  <node name="JointTrajectoryActionServer" pkg="intera_interface" type="joint_trajectory_action_server.py" output="screen"/>
  <node name="sawyer_movement_server" pkg="the_mighty_sawyer" type="sawyer_movement_server" output="screen"/>

  <!-- Launch State Machine -->
  <node name="sawyer_main_client" pkg="the_mighty_sawyer" type="sawyer_main_client" output="screen"/>
</launch>
